File: drills/ts/harnesses/nc64_climbing_stairs.ts
import * as solution from "./solution";

const show = (v: unknown) => JSON.stringify(v) ?? "undefined";

export function run(): [string, string, string][] {
  if (typeof solution.climbStairs !== "function") throw new Error("__signature_mismatch__");
  return [
    ["climbStairs(2)", show(2), show(solution.climbStairs(2))],
    ["climbStairs(3)", show(3), show(solution.climbStairs(3))],
    ["climbStairs(1)", show(1), show(solution.climbStairs(1))],
    ["climbStairs(0)", show(1), show(solution.climbStairs(0))],
    ["climbStairs(10)", show(89), show(solution.climbStairs(10))],
    ["climbStairs(45)", show(1836311903), show(solution.climbStairs(45))],
  ];
}

File: drills/go/harnesses/nc64_climbing_stairs.go
package main

func main() {
	run(func() []testCase {
		return []testCase{
			tc("climbStairs(2)", 2, climbStairs(2)),
			tc("climbStairs(3)", 3, climbStairs(3)),
			tc("climbStairs(1)", 1, climbStairs(1)),
			tc("climbStairs(10)", 89, climbStairs(10)),
			tc("climbStairs(45)", 1836311903, climbStairs(45)),
		}
	})
}
